pkg/prometheus/promcfg_test.go

```diff
@@ -884,6 +893,7 @@ func TestK8SSDConfigGeneration(t *testing.T) {
 				Spec: monitoringv1.PrometheusSpec{
 					CommonPrometheusFields: monitoringv1.CommonPrometheusFields{
 						IgnoreNamespaceSelectors: false,
+						ServiceDiscoveryRole:     "EndpointSlice",
```

Comment:
Thank you for pointing this out. I have removed this setting in [5cab2fc](https://github.com/prometheus-operator/prometheus-operator/pull/6672/commits/5cab2fc5481a9816f3adfeb919c8353435322c55)

I have opted not to add `assert.False(t, cg.endpointSliceSupported)` as it will default to false.